tool/tctl/common/loadtest_command.go

```diff
@@ -126,18 +128,30 @@ func (c *LoadtestCommand) NodeHeartbeats(ctx context.Context, client *authclient
 		fmt.Fprintf(os.Stderr, "[!] "+format+"\n", args...)
 	}
 
-	infof("Setting up node hb load generation. count=%d, churn=%d, labels=%d, interval=%s, ttl=%s, concurrency=%d",
+	durationReadable := c.duration.String()
+	if c.duration == 0 {
+		durationReadable = "forever"
+	}
+
+	infof("Setting up node hb load generation. count=%d, churn=%d, labels=%d, interval=%s, ttl=%s, duration=%s, concurrency=%d",
 		c.count,
 		c.churn,
 		c.labels,
 		c.interval,
 		c.ttl,
+		durationReadable,
 		c.concurrency,
 	)
 
 	ctx, cancel := signal.NotifyContext(ctx, os.Interrupt)
 	defer cancel()
 
+	if c.duration > 0 {
+		var stop context.CancelFunc
+		ctx, stop = context.WithTimeout(ctx, c.duration)
+		defer stop()
```

Comment:
nit, style and consistency (we tend to use `cancel` everywhere where we create a subcontext, and the returned function isn't really "stopping" the timeout, it's more like it's fast-forwarding it)

```suggestion
		var cancel context.CancelFunc
		ctx, cancel = context.WithTimeout(ctx, c.duration)
		defer cancel()
```

plus personal preference (either way is fine, I just prefer pushing values out into the surrounding scope explicitly)

```suggestion
		timeoutCtx, cancel := context.WithTimeout(ctx, c.duration)
		defer cancel()
		ctx = timeoutCtx
```

Reply:
Thanks! Makes sense, I will update.